src/NServiceBus.Core/Recoverability/DefaultRecoverabilityPolicy.cs

```diff
@@ -18,9 +18,13 @@ public static class DefaultRecoverabilityPolicy
         /// <returns>The recoverability action.</returns>
         public static RecoverabilityAction Invoke(RecoverabilityConfig config, ErrorContext errorContext)
         {
-            if (errorContext.Exception is MessageDeserializationException)
+            // ReSharper disable once LoopCanBeConvertedToQuery
+            foreach (var unrecoverableExceptionType in config.Failed.UnrecoverableExceptionTypes)
```

Comment:
Too bad we can't just do `config.Failed.UnrecoverableExceptionTypes.Contains(errorContext.Exception.GetType())` here. How important do you think it is to be considering the inheritance hierarchy vs. just making the collection be an exact match on type?